In [1]:
# Importing required libraries
import pandas as pd

In [2]:
# Loading our attrition data into a pandas dataframe
df = pd.read_csv(
    "../data/raw/WA_Fn-UseC_-HR-Employee-Attrition.csv"
)

# Visualizing top 5 records
df.head()

,Age,Attrition,BusinessTravel,DailyRate,Department,DistanceFromHome,Education,EducationField,EmployeeCount,EmployeeNumber,...,RelationshipSatisfaction,StandardHours,StockOptionLevel,TotalWorkingYears,TrainingTimesLastYear,WorkLifeBalance,YearsAtCompany,YearsInCurrentRole,YearsSinceLastPromotion,YearsWithCurrManager
0,41,Yes,Travel_Rarely,1102,Sales,1,2,Life Sciences,1,1,...,1,80,0,8,0,1,6,4,0,5
1,49,No,Travel_Frequently,279,Research & Development,8,1,Life Sciences,1,2,...,4,80,1,10,3,3,10,7,1,7
2,37,Yes,Travel_Rarely,1373,Research & Development,2,2,Other,1,4,...,2,80,0,7,3,3,0,0,0,0
3,33,No,Travel_Frequently,1392,Research & Development,3,4,Life Sciences,1,5,...,3,80,0,8,3,3,8,7,3,0
4,27,No,Travel_Rarely,591,Research & Development,2,1,Medical,1,7,...,4,80,1,6,3,3,2,2,2,2


In [3]:
# Check for completely duplicated rows
duplicate_rows = df.duplicated().sum()
print("Number of duplicated rows:", duplicate_rows)

Number of duplicated rows: 0


In [4]:
# Check duplicates using the EmployerNumber column (Unique Identifier)
duplicate_employee_numbers = df["EmployeeNumber"].duplicated().sum()
print("Duplicated EmployeeNumbers:", duplicate_employee_numbers)

Duplicated EmployeeNumbers: 0


In [5]:
# Summary
print("Total rows:", len(df))
print("Completely duplicated rows:", df.duplicated().sum())
print("Unique EmployeeNumbers:", df["EmployeeNumber"].nunique())
print("Duplicated EmployeeNumbers:", df["EmployeeNumber"].duplicated().sum())

Total rows: 1470
Completely duplicated rows: 0
Unique EmployeeNumbers: 1470
Duplicated EmployeeNumbers: 0


#### **Conclusion**

- The duplicate analysis confirms that the dataset contains **1,470 records with no completely duplicated rows**. In addition, all **1,470 EmployeeNumbers are unique**, with no duplicated employee identifiers identified. These findings indicate that each record represents a distinct employee and that there are no apparent duplicate records requiring removal based on the checks performed. The dataset can therefore proceed to the next stage of data preparation and analysis.
